### PR TITLE
Add vapp from template + fixes

### LINKED ISSUE #228: Remove current Vdc.addVapp method and add Vdc.addVappFromTemplate

### PR DESCRIPTION
**Pull request by JoMas971**
Resolves #228

### CODE REVIEW

File: src/sdk/model/common/__json__/fence-mode-request-type.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type FenceModeRequestType = 'BRIDGED' |
```
Comment (cfsnyder) [inline src/sdk/model/common/__json__/fence-mode-request-type.ts]: Remove "Request" from the name

Comment (JoMas971) [reply src/sdk/model/common/__json__/fence-mode-request-type.ts]: PR updated ;-)

File: src/sdk/model/common/__json__/network-fence-mode-type.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/**
```
Comment (cfsnyder) [inline src/sdk/model/common/__json__/network-fence-mode-type.ts]: This is the same as the fence-mode-type, so we should get rid of this one

**Comment by JoMas971** [comment]
@cfsnyder I've fixed all your comments. It should be good now.

File: src/sdk/model/vapp/__json__/vapp-network-initialization-params-request-json.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { IpRangeJson } from '../../common/ip-range/__json__/ip-range-json';
+
+export interface VappNetworkInitializationParamsRequestJson {
```
Comment (cfsnyder) [inline src/sdk/model/vapp/__json__/vapp-network-initialization-params-request-json.ts]: is this actually used anywhere now?

File: src/sdk/model/vapp/vapp-network-initialization-params-request-json.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
+import { VappNetworkInitializationParamsRequestJson } from './__json__/vapp-network-initialization-params-request-json';
+import { IpRange } from '../common/ip-range/ip-range';
+import { IpRangeJson } from '../common/ip-range/__json__/ip-range-json';
+
+/**
+ * Vapp network initialization params request.
+ */
+/* istanbul ignore next: autogenerated */
+export class VappNetworkInitializationParamsRequest {
```
Comment (cfsnyder) [inline src/sdk/model/vapp/vapp-network-initialization-params-request-json.ts]: actually used anywhere now?

File: src/sdk/model/vdc/add-vapp-request.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import { VappNetworkInitializationParamsRequestJson } from
+    '../vapp/__json__/vapp-network-initialization-params-request-json';
+import { VappNetworkInitializationParamsRequest } from '../vapp/vapp-network-initialization-params-request-json';
+import { VmCreateRequest } from '../vm/vm-create-request';
+
+/**
+ * Add vapp request.
+ */
+/* istanbul ignore next: autogenerated */
+export class AddVappRequest {
```
Comment (cfsnyder) [inline src/sdk/model/vdc/add-vapp-request.ts]: used anywhere now?

File: src/sdk/model/vapp/__json__/vapp-network-initialization-params-request-json.ts (hunk 3 above)
Comment (JoMas971) [reply src/sdk/model/vapp/__json__/vapp-network-initialization-params-request-json.ts]: indeed you're right. I can get rid of the old methods. I'll change this.

**Comment by JoMas971** [comment]
Ok, this time i think it's good.

**Pull request merged**